### DISCUSSION TITLE
p/e124cb73d5864c24bb5547cd3431e338/

### DISCUSSION THREAD

**Discussion by giscus** [rxliuli/blog#56] [Announcements]
# p/e124cb73d5864c24bb5547cd3431e338/

场景最近迁移了一些 API，因为所有前端项目都在同一个 monorepo 中，所以作为 lib 维护者吾辈还需要帮助迁移其他使用的模块。由于项目数量较多（大约有 30 多个），手动迁移非常麻烦而且难以测试。所以在调研了一些现有的大规模重构的方法后，吾辈选择了 jscodeshift 作为主要工具来做自动化迁移。那么，它相比于使用 ide 的重构功能、使用字符串搜索替换亦或是手工一个个替换有什么不同

https://blog.rxliuli.com/p/e124cb73d5864c24bb5547cd3431e338/

**Comment by R4DEONN** [comment]
Hello) Thank you for your article. It was very useful to me.
